pkg/provider/vault/client_get.go

```diff
@@ -227,6 +227,9 @@ func (c *client) buildMetadataPath(path string) (string, error) {
 			url = fmt.Sprintf("%s/metadata/%s", *c.store.Path, path)
 		}
 	}
+	if c.namespace != "" {
```

Comment:
This `c.namespace` is the Kubernetes namespace of the ExternalSecret. It is _not_ related to the Vault Enterprise namespace whatsoever. 

In the initial implementation it was set properly, it used to work at some point (this is ~3 years ago):
https://github.com/external-secrets/external-secrets/blob/7be249ba6327cde0e4e4cf22e6ab9a87be40c0d5/pkg/provider/vault/vault.go#L121-L123

It was probably lost in some refactoring and due to the lack of e2e tests for this enterprise feature we can't guarantee it is working reliably :disappointed:

Reply:
Ah thank you. I was pretty sure I'm confusing something here. 🤣

Reply:
I think this is the PR that introduced this..? https://github.com/external-secrets/external-secrets/commit/731c0ed736894582a7a21d257decb0cdceefe4d7